drivers/hmis/app/jobs/hmis/auto_exit_job.rb

```diff
@@ -0,0 +1,80 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Hmis
+  class AutoExitJob < BaseJob
+    def perform
+      Hmis::Hud::Project.all.each do |project|
+        config = Hmis::AutoExitConfig.config_for_project(project)
+        next unless config.present?
+
+        project.enrollments.each do |enrollment|
+          most_recent_contact = if HudUtility2024.residential_project_type_ids.include?(project.project_type)
+            enrollment.services.where(record_type: 200).order(:date_provided).last
+          else
+            [
+              enrollment.services.order(:date_provided).last,
+              enrollment.custom_services.order(:date_provided).last,
+              enrollment.current_living_situations.order(:information_date).last,
+              enrollment.custom_assessments.order(:assessment_date).last,
+            ].compact.max_by { |entity| contact_date_for_entity(entity) }
+          end
+
+          next unless most_recent_contact.present?
```

Comment:
This shouldn't really happen, because every enrollment should at least have an intake assessment. If it does, though, let's use the `enrollment.entry_date` as the `most_recent_contact_date` instead. That way, if someone enters and doesn't have _anything_ recorded for 3 months (or whatever), they would still get auto-exited after 3 months.